model/postgres/postgres_server.rb

```diff
@@ -95,14 +98,22 @@ def check_pulse(session:, previous_pulse:)
     session[:db_connection] ||= Sequel.connect(adapter: "postgres", host: health_monitor_socket_path, user: "postgres")
 
     reading = begin
-      session[:db_connection]["SELECT 1"].all && "up"
+      lsn_function = primary? ? "pg_current_wal_lsn()" : "pg_last_wal_receive_lsn()"
+      lsn_monitor.last_known_lsn = session[:db_connection]["SELECT #{lsn_function} AS lsn"].first[:lsn]
+      "up"
     rescue
       "down"
     end
-    pulse = aggregate_readings(previous_pulse: previous_pulse, reading: reading)
+    pulse = aggregate_readings(previous_pulse: previous_pulse, reading: reading, data: {last_known_lsn: lsn_monitor.last_known_lsn})
+
+    DB.transaction do
+      if pulse[:reading] == "up" && pulse[:reading_rpt] % 12 == 1
```

Comment:
I feel these modulus and constants have a subtle connection with the period of monitoring generally, can we somehow couple them in more obvious form? This may be as simple as some named constants, or a module, or whatever (I don't have faith in my ability to forecast the exact code form of the coupling).